src/ucp/wireup/wireup.c

```diff
@@ -1758,6 +1713,66 @@ ucs_status_t ucp_wireup_connect_remote(ucp_ep_h ep, ucp_lane_index_t lane)
     return status;
 }
 
+static uct_ep_h ucp_wireup_get_tl_ep(uct_ep_h uct_ep)
+{
+    ucp_wireup_ep_t *wireup_ep = ucp_wireup_ep(uct_ep);
+    return (wireup_ep == NULL) ? uct_ep : wireup_ep->super.uct_ep;
+}
+
+static void
+ucp_wireup_fill_is_connected_params(const ucp_address_entry_t *addr_entry,
+                                    uct_ep_is_connected_params_t *params)
+{
+    params->field_mask = 0;
+    if (addr_entry->dev_addr != NULL) {
+        params->device_addr = addr_entry->dev_addr;
+        params->field_mask |= UCT_EP_IS_CONNECTED_FIELD_DEVICE_ADDR;
+    }
+
+    if (addr_entry->iface_addr != NULL) {
+        params->iface_addr  = addr_entry->iface_addr;
+        params->field_mask |= UCT_EP_IS_CONNECTED_FIELD_IFACE_ADDR;
+    }
+}
+
+static ucp_lane_index_t
+ucp_wireup_get_remote_lane(uct_ep_h uct_ep,
+                           const ucp_address_entry_t *addr_entry)
+{
+    uct_ep_is_connected_params_t params;
+    const ucp_address_entry_ep_addr_t *ep_entry;
+
+    ucp_wireup_fill_is_connected_params(addr_entry, &params);
+
+    ucs_carray_for_each(ep_entry, addr_entry->ep_addrs,
+                        addr_entry->num_ep_addrs) {
+        params.field_mask |= UCT_EP_IS_CONNECTED_FIELD_EP_ADDR;
+        params.ep_addr     = ep_entry->addr;
+        if (uct_ep_is_connected(ucp_wireup_get_tl_ep(uct_ep), &params)) {
+            return ep_entry->lane;
+        }
+    }
+
+    return UCP_NULL_LANE;
+}
+
+int ucp_wireup_is_lane_connected(ucp_ep_h ep, ucp_lane_index_t lane,
+                                 const ucp_address_entry_t *addr_entry)
+{
+    uct_ep_h uct_ep = ucp_ep_get_lane(ep, lane);
+    uct_ep_is_connected_params_t params;
+
+    uct_ep = ucp_wireup_get_tl_ep(uct_ep);
+    ucp_wireup_fill_is_connected_params(addr_entry, &params);
+
+    if ((addr_entry->num_ep_addrs == 0) &&
+        uct_ep_is_connected(uct_ep, &params)) {
```

Comment:
before doing this, need to check that the relevant UCT transport (iface) supports to-iface connection

Reply:
replacing this check with to-iface connection support check is erroneous. 
The real test should be whether uct_ep is **connected** by iface or by EP.
But since we don't have such UCT_API, the only valid test is the current one.